providers/google/src/airflow/providers/google/cloud/hooks/gen_ai.py

```diff
@@ -194,3 +197,182 @@ def create_cached_content(
         )
 
         return resp.name
+
+
+class GenAIGeminiAPIHook(GoogleBaseHook):
+    """Class for Google Cloud Generative AI Gemini Developer API hook."""
+
+    def __init__(self, gemini_api_key: str, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.gemini_api_key = gemini_api_key
+
+    def get_genai_client(self):
+        return genai.Client(
+            api_key=self.gemini_api_key,
+            vertexai=False,
+        )
+
+    def get_batch_job(
+        self,
+        job_name: str,
+    ) -> BatchJob:
+        """
+        Get batch job using Gemini Batch API.
+
+        :param job_name: Required. Batch job name.
+        """
+        client = self.get_genai_client()
+        resp = client.batches.get(name=job_name)
+        return resp
+
+    def list_batch_jobs(
+        self,
+        list_batch_jobs_config: ListBatchJobsConfig | dict | None = None,
+    ):
```

Comment:
Could we specify the return type as we have in Client https://github.com/googleapis/python-genai/blob/main/google/genai/batches.py#L2009 ?

Reply:
done